File: Binary Calculation/Binary Subtraction/BinarySub.py
def binAddition( a, b):
    while (b != 0): 
        c = (a & b) << 1
        a=a^b
        b=c
    return a

def binSubtracton(a, b):
    b = binAddition(~b, 1)
    while (b != 0):
        carry = (a & b) << 1
        a = a ^ b
        b = carry
    return a

n1=int(input("Input first integer value: "))
n2=int(input("Input second integer value: "))
binSub=binSubtracton(n1,n2)
print("Binary Subtraction: {}".format(binSub))

File: Binary Calculation/Binary Subtraction/BinarySub.cpp
#include<iostream>
using namespace std;
int binAddition(int a,int b){
      int c;
      while (b != 0) {
              c = (a & b) << 1;
              a=a^b;
              b=c;
      }
      return a;
}
int binSubtracton(int a, int b){
      int carry;
      b = binAddition(~b, 1);
      while (b != 0) {
              carry = (a & b) << 1;
              a = a ^ b;
              b = carry;
      }
      return a;
}
void main(){
	  int number1,number2,binSub;

	  cout<<"Input first integer value: ";
	  cin>>number1;
	
	  cout<<"Input second integer value: ";
	  cin>>number2;
	  binSub=binSubtracton(number1,number2);

	  cout<<"Binary Subtraction: "<<binSub;
}